src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -113,37 +113,46 @@ public static Complex Divide(Complex dividend, Complex divisor)
 
         public static double Abs(Complex value)
         {
-            if (double.IsInfinity(value._real) || double.IsInfinity(value._imaginary))
-            {
-                return double.PositiveInfinity;
-            }
-
             return Hypot(value._real, value._imaginary);
         }
 
         private static double Hypot(double a, double b)
         {
-            // |value| == sqrt(a^2 + b^2)
-            // sqrt(a^2 + b^2) == a/a * sqrt(a^2 + b^2) = a * sqrt(a^2/a^2 + b^2/a^2)
-            // Using the above we can factor out the square of the larger component to dodge overflow.
+            // Using
+            //   sqrt(a^2 + b^2) = |a| * sqrt(a^2/a^2 + b^2/a^2)
+            // we can factor out the larger component to dodge overflow even when a * a would overflow.
 
-            double c = Math.Abs(a);
-            double d = Math.Abs(b);
+            a = Math.Abs(a);
+            b = Math.Abs(b);
+
+            double small, large;
+            if (a < b)
+            {
+                small = a;
+                large = b;
+            }
+            else
+            {
+                small = b;
+                large = a;
+            }
 
-            if (c > d)
+            if (small == 0.0)
             {
-                double r = d / c;
-                return c * Math.Sqrt(1.0 + r * r);
+                return (large);
             }
-            else if (d == 0.0)
+            else if (Double.IsPositiveInfinity(large) && !Double.IsNaN(small))
```

Comment:
nit: Can you replace `Double` with `double` everywhere? We prefer to use the C# keywords everywhere, even in static functions.